tests/migrations/upgrade_test.py

```diff
@@ -303,12 +339,398 @@ def test_upgrade_v23_6_to(
+            id=5,
+            token_address="token_address2",
+            exchange_address="exchange_address2",
+            modified=datetime(2023, 4, 6, 0, 0, 0),
+        )  # remains after migration
+
+        with engine.connect() as conn:
+            conn.execute(stmt1)
+            conn.execute(stmt2)
+            conn.execute(stmt3)
+            conn.execute(stmt4)
+            conn.execute(stmt5)
+            conn.commit()
+
+        # NOTE: idx_position_membership_block_number data
+        idx_position_membership_block_number = meta.tables.get(
+            "idx_position_membership_block_number"
+        )
+        stmt1 = insert(idx_position_membership_block_number).values(
+            id=1,
+            token_address="token_address1",
+            exchange_address="exchange_address1",
+            modified=datetime(2023, 4, 5, 0, 0, 0),
+        )  # removed after migration
+        stmt2 = insert(idx_position_membership_block_number).values(
+            id=2,
+            token_address="token_address1",
+            exchange_address="exchange_address1",
+            modified=datetime(2023, 4, 6, 0, 0, 0),
+        )  # remains after migration
+        stmt3 = insert(idx_position_membership_block_number).values(
+            id=3,
+            token_address="token_address1",
+            exchange_address="exchange_address2",
+            modified=datetime(2023, 4, 6, 0, 0, 0),
+        )  # remains after migration
+        stmt4 = insert(idx_position_membership_block_number).values(
+            id=4,
+            token_address="token_address2",
+            exchange_address="exchange_address2",
+            modified=datetime(2023, 4, 5, 0, 0, 0),
+        )  # removed after migration
+        stmt5 = insert(idx_position_membership_block_number).values(
+            id=5,
+            token_address="token_address2",
+            exchange_address="exchange_address2",
+            modified=datetime(2023, 4, 6, 0, 0, 0),
+        )  # remains after migration
+
         with engine.connect() as conn:
             conn.execute(stmt1)
             conn.execute(stmt2)
+            conn.execute(stmt3)
+            conn.execute(stmt4)
+            conn.execute(stmt5)
+            conn.commit()
 
         # 3. Run to head
         alembic_runner.migrate_up_to("head")
+
+        with engine.connect() as conn:
+            # NOTE: position
+            all_row_count = conn.execute(text("SELECT COUNT(*) FROM position")).scalar()
+            token_address_1_count = conn.execute(
+                text(
+                    "SELECT COUNT(*) FROM position WHERE token_address = 'token_address1'"
+                )
+            ).scalar()
+
+            # Ensure that there are 3 records.
+            #    (token_address1, account_address1)
+            #    (token_address1, account_address2)
+            #    (token_address2, account_address2)
+            assert all_row_count == 3
+            assert token_address_1_count == 2
```

Comment:
- Reproduce datas that may already exists and assert the results as correct after migration.